apm/otel_monitor.go

```diff
@@ -483,7 +483,9 @@ func stripValue(val bson.RawValue) (bson.RawValue, error) {
 		}
 		arr = compactArray(arr)
 		_, encodedArray, err := bson.MarshalValue(arr)
-		return bson.RawValue{Type: bson.TypeArray, Value: encodedArray}, errors.Wrap(err, "encoding array")
+		copiedArr := make([]byte, len(encodedArray))
```

Comment:
This was the cause of the failing test. This is a bug in the go driver, where it will override the underlying []byte when MarshalValue is called in quick succession. I've started a thread with them, but there's no ticket tracking this right now. I'll create a ticket linking to their ticket saying to stop copying.

I'm not sure if this can happen with other MarshalValue calls like below for string or above for documents, I could only confirm for arrays.

Reply:
Can we add a comment on the code to this effect? It's really niche and seemingly a bug on the Go driver's end, so clarifying why this is needed would help us figure out what it's for + if we still need it in the future.

Reply:
This bug was solved in v2.0.1, so I reverted it and we don't need to handle it 🚀